brokers/unified/broker.go

```diff
@@ -191,22 +191,28 @@ func validateMetas(metas []model.PluginMeta) error {
 // defaultRegistry is used as the registry when plugin does not specify its registry.
 // If defaultRegistry is empty, and plugin does not specify a registry, an error is returned.
 func (b *Broker) GetPluginMeta(plugin model.PluginFQN, defaultRegistry string) (*model.PluginMeta, error) {
-	log.Printf("Fetching plugin meta.yaml for %s", plugin.ID)
-	registry, err := getRegistryURL(plugin, defaultRegistry)
-	if err != nil {
-		return nil, err
+	var pluginURL string
+	if plugin.Reference != "" {
+		pluginURL = plugin.Reference
+		log.Printf("Fetching plugin meta.yaml from reference %s", pluginURL)
+	} else {
+		log.Printf("Fetching plugin meta.yaml for %s", plugin.ID)
```

Comment:
I think it makes sense to log registry URL here as well.
Or have single log for both cases (reference, registry): `Fetching plugin meta.yaml from %s`, pluginURL

Reply:
makes sense